src/Single.php

```diff
@@ -130,6 +130,64 @@ public static function filterTrue(iterable $data, callable $predicate = null): \
         }
     }
 
+    /**
+     * Filter out elements from the iterable only returning unique elements.
+     *
+     * If $strict is true:
+     *  - scalars: compares strictly by type;
+     *  - objects: always treats different instances as not equal to each other;
+     *  - arrays: compares serialized.
+     *
+     * If $strict is false:
+     *  - scalars: compares non-strictly by value;
+     *  - objects: compares serialized;
+     *  - arrays: compares serialized.
+     *
+     * @template T
+     * @param iterable<T> $data
+     * @param bool $strict
+     *
+     * @return \Generator<T>
+     */
+    public static function filterUnique(iterable $data, bool $strict = true): \Generator
```

Comment:
I like this functionality and think it is a good addition. However, this is a bit complicated due to PHP's loose vs strict comparisons. I'd like to spend more time thinking about this and what the expected behavior should be. How about removing `filterUnique` from this pull request, and submit it again as a separate pull request with just that functionality? This way it will not block the rest of this pull request. Thanks.

Reply:
Here is a new pull request for this method: https://github.com/markrogoyski/itertools-php/pull/4